### ISSUE TITLE
PG error when try to only show last 24 hours data using group_by_hour with chartkick

### ISSUE THREAD

**Issue by Rubioli** [ankane/groupdate#171]
I use `chartkick gem`, I only want to show data for **last 24 hours**. I did it by doing:
`= area_chart Visit.group_by_hour(:post_id, last: 24).count`

This works fine in my `Development ENV`, where I have `sqlite3`. In my `Production ENV` I use *PostgreSql (`pg gem`)*.

In my view, where I have my charts *(In `Production ENV`)*, I get this error:

```
PG::CannotCoerce: ERROR:  cannot cast type integer to timestamp with time zone
LINE 1: ...OUNT(*) AS count_all, (DATE_TRUNC('hour', (post_id::timestam...
                                                             ^
: SELECT COUNT(*) AS count_all, (DATE_TRUNC('hour', (post_id::timestamptz - INTERVAL '0 second') AT TIME ZONE 'Etc/UTC') + INTERVAL '0 second') AT TIME ZONE 'Etc/UTC' AS hour FROM "visits" WHERE (post_id >= '2017-05-07 12:00:00.000000' AND post_id <= '2017-05-08 11:18:48.083853') AND "visits"."influencer" = 1 GROUP BY (DATE_TRUNC('hour', (post_id::timestamptz - INTERVAL '0 second') AT TIME ZONE 'Etc/UTC') + INTERVAL '0 second') AT TIME ZONE 'Etc/UTC'
```
On `= area_chart Visit.group_by_hour(:post_id, last: 24).count`

How can I only show last 24 hours data and doesn't get error when having *PostgreSql*?

*I am using `Rails 4.2.4`, `Ahoy gem` & `chartkick gem` as well.*

Thanks in advanced!

**Comment by mccallumjack** [comment]
`post_id` doesn't sound like a timestamp, are you sure that is what you are trying to group by?

Looks like it is an integer, so you get:
```
PG::CannotCoerce: ERROR:  cannot cast type integer to timestamp with time zone
```
you'll want to group by a timestamp, like `created_at`.

**Comment by Rubioli** [comment]
Thanks @mccallumjack.

Yes, it makes sense. Only thing that made me to think that, it was correct was because it worked on development.

Now I have changed it and added `started_at` and it works, but now Im confuse how to get it right.

For instance, I used:
`= area_chart Visit.group_by_hour(:post_id, last: 24).where(user: current_user).count` to get data for **clicks sent per post  last 24 hrs** by this user and my chart had `post_id` as `x` and clicks as `y`.

Where is it I can add `post_id` here?

Thanks in advanced!
